subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -1186,17 +1234,35 @@ static int bt_bap_scan_delegator_init(void)
 #endif /* CONFIG_BT_BAP_SCAN_DELEGATOR_RECV_STATE_COUNT > 2 */
 #endif /* CONFIG_BT_BAP_SCAN_DELEGATOR_RECV_STATE_COUNT > 1 */
 
-	bt_le_per_adv_sync_cb_register(&pa_sync_cb);
+	if (!pa_sync_cb_registered) {
+		bt_le_per_adv_sync_cb_register(&pa_sync_cb);
```

Comment:
Missing error handling of `bt_le_per_adv_sync_cb_register`

Reply:
Is it really needed? because bt_le_per_adv_sync_cb_register only returns 0 (on success) and EEXIST (if already registered, which is not important or never happens here). I also can see that it hasn’t been handled in any other places…

Reply:
If it can only fail in the case of programming errors, then we should add an ASSERT IMO. 

The return value was recently added which would explain why it isn't being checked everywhere, but it should.

Reply:
Ok, fair enough. I’ll prefer to handle it by returning an error instead of using ASSERT, as more errors might be implemented in the future and the probable error is not fatal...